### PR TITLE
Python 3 fixes.

### PR DESCRIPTION
**Pull request by pscadding**
Python 3 port.
Fairly small changes, most of the changes are on the `tk-framework-adobe` repo.

### CODE REVIEW

File: startup.py [hunk 1]
```diff
@@ -81,13 +81,11 @@ def prepare_launch(self, exec_path, args, file_to_open=None):
...
-            self.disk_location, "python", "startup", "bootstrap.py"
+        bootstrap_python_path = os.path.join(self.disk_location, "python", "startup")
+        (file_obj, filename, desc) = imp.find_module(
+            "bootstrap", [bootstrap_python_path]
         )
-        with open(bootstrap_python_path, "r") as mod_file:
-            bootstrap = imp.load_module(
-                "bootstrap", mod_file, "tk-photoshopcc.bootstrap", (".py", "U", 1)
-            )
+        bootstrap = imp.load_module("bootstrap", file_obj, filename, desc)
```
Comment (pscadding) [inline startup.py]: This was erroring on Python 3 as it couldn't find the filename that it was passed.
I've updated it to get the details from find_module, but this is still using the deprecated `imp`. 

I did consider sys.path.insert and then removing the path after importing, as the alternatives to `imp` don't appear to be that backwards compatible.

File: engine.py [hunk 2]
```diff
@@ -724,7 +725,7 @@ def _handle_active_document_change(self, active_document_path):
         # Make sure we have a properly-encoded string for the path. We can
         # possibly get a file path/name that contains unicode, and we don't
         # want to deal with that later on.
-        if isinstance(active_document_path, unicode):
+        if isinstance(active_document_path, six.text_type):
             active_document_path = active_document_path.encode("utf-8")
```
Comment (jfboismenu) [inline engine.py]: This will fail in Python 3, as encode does not exist on `str`. You need to call six.ensure_str

File: startup.py (hunk 1 above)
Comment (jfboismenu) [reply startup.py]: Could you use the `sys.path.insert` method instead, so we don't have to revisit this when `imp` goes away?

Comment (pscadding) [reply startup.py]: As discussed I've duplicated up our internal core imp ticket and made it about Photoshop.

File: engine.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
 
 import sgtk
 from sgtk.util.filesystem import ensure_folder_exists
+from tank_vendor import six
```
Comment (jfboismenu) [inline engine.py]: Please update the minimum core version for the engine in info.yml.

**Pull request merged**